### ISSUE TITLE
issue w/ .count().where() + "OR" modifier

### ISSUE THREAD

**Issue by mikermcneil** [balderdashy/waterline#449]
The following query generates an error from `Deferred` in WL core:

```js
Foo.count().where({
  or: [
    { /* ... */ },
    { /* ... */ }
  ]
})
.exec(...)
```

It can be worked around by changing the query to:

```js
Foo.count({
  where: {
    or: [
      { /* ... */ },
      { /* ... */ }
    ]
  }
})
.exec(...)
```

@particlebanana Good news is this appears to be an issue w/ query normalization in `.count()`, which means it's easy to apprehend.  Unfortunately, it also means the "find" engine rewrite won't nab it and we'll have to deal with it separately.  So not a huge immediate deal since there's a workaround, but I want to make sure and track it here

**Comment by devinivy** [comment]
Is this still kicking around?

**Comment by sailsbot** [comment]
Thanks for posting, @mikermcneil.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
